### PR TITLE
Use message send checkpoints

### PR DESCRIPTION
**Pull request by sumnerevans**
Requires https://github.com/mautrix/python/pull/66

### CODE REVIEW

File: mautrix_facebook/portal.py [hunk 1]
```diff
@@ -573,7 +574,22 @@ async def handle_matrix_message(self, sender: 'u.User', message: MessageEventCon
...
+                MessageSendCheckpointStatus.PERM_FAILURE,
+                event_id,
+                self.mxid,
+                EventType.ROOM_MESSAGE,
+                message.msgtype,
+                error=e,
+            )
+            await self._send_bridge_error(str(e))
+            self.log.exception("Failed to handle matrix message.", e)
+            raise
```
Comment (tulir) [inline mautrix_facebook/portal.py:590]: This probably doesn't need to be raised further if we already log and checkpoint it

File: mautrix_facebook/portal.py [hunk 2]
```diff
@@ -573,7 +574,22 @@ async def handle_matrix_message(self, sender: 'u.User', message: MessageEventCon
                                     event_id: EventID) -> None:
         # TODO handle errors?
         # try:
```
Comment (tulir) [inline mautrix_facebook/portal.py:576]: This TODO and the corresponding commented `except` block below are done now

**Review by tulir — APPROVED** [review]
Mostly fine but a few lines could be removed

File: mautrix_facebook/portal.py (hunk 2 above)
Comment (sumnerevans) [reply mautrix_facebook/portal.py:576]: It seems like they are dealing with specific error types? Is there any benefit to keeping that around at all?

Comment (tulir) [reply mautrix_facebook/portal.py:576]: No, it's legacy fbchat stuff (pre mobile api rewrite)

**Pull request merged**